CMakeLists.txt

```diff
@@ -123,6 +123,18 @@ else ()
     message (AUTHOR_WARNING "${ISSUE_REPORT_MSG}")
 endif ()
 
+# MSVC symbol visibility and linker behavior is sufficiently different from all
+# other platforms. Here, we ensure tests/tools built as EXE choose symbol
+# visibility appropriate for linking the library, whether static or dynamic.
+if (MSVC)
```

Comment:
This would actually break mingw-w64 (& non-msvc in general) builds because NN_EXPORT isn't defined to `__declspec(dllimport)`/`__declspec(dllexport)` on non-msvc compilers anymore. CMake solves this problem when generating an export header, but I would rather have it leave NN_EXPORT as a conditional of `_WIN32`, not `NN_HAVE_MSVC` (or maybe let it use `__attribute__((dllimport))` on GNU compilers).